app.py

```diff
@@ -0,0 +1,278 @@
+                else:
+                    cprint('Error! type living or office!')
+                    return
+
+        except ValueError as e:
+            cprint(e, 'red')
+
+    @pass_opt
+    def do_add_person(self, arg):
+        """
+        Creates a person and assign them to an office, living space or both. 
+
+        Usage:
+            add_person <first_name> <last_name> [--accomodation=n]
+        """
+        try:
+            person = arg["<first_name>"] + " " + arg["<last_name>"]
+            wants_accomodation = arg["--accomodation"]
+            print ("Name: "+person)
+            
+            person_type = input("Enter a role, either 'staff' or 'fellow': ")
+            while person_type.lower() not in ['staff','fellow']:
+                print('Invalid Role')
+                person_type = input("Enter a role, either 'staff' or 'fellow': ")
+
+            if person_type.lower() == 'fellow':
+                if wants_accomodation == None:
+                    wants_accomodation = 'n'
+                elif wants_accomodation.lower() in ["yes", "y", "no", 'n']: 
+                    print ("Wants Accomodation: "+wants_accomodation)
+                else:
+                    cprint('Invalid Choice for wants accommodation', 'red')
+                    return
+            elif person_type.lower() == 'staff':
+                if wants_accomodation != None and wants_accomodation.upper() == 'Y':
+                    cprint('\n Staff members can not be allocated accommodation', 'green')
+                wants_accomodation = 'n'
+
+            amity.create_person({"person_name": person, "role": person_type, "wants_accomodation": wants_accomodation})
+        except ValueError as e:
+            cprint((e), 'red')
+
+    @pass_opt
+    def do_reallocate_person(self, arg):
+        """
+        Reallocates a person from their current room to another room.
+
+        Usage:
+            reallocate_person <first_name> <last_name> 
+        """
+        try:
+            person_name = arg["<first_name>"] + " " + arg["<last_name>"]
+            person = self.get_person(person_name)
+            if person:
+                print("Name: "+person.person)
+                new_room = input("Enter the name of the new room: ")
+                amity.reallocate_person(person, new_room) 
+            else:
+                cprint('Such a person does not exist!', 'red')
+        except ValueError as e:
+            cprint(e, 'red')
+
+  
+    def get_person(self, person_name):
+        '''
+        Checks whether the person exists in the list of people
+        '''
+        try:
+            names = [person.person for person in amity.all_people]
+            if person_name not in names: return False
+            else: 
+                person = [person for person in amity.all_people if person.person == person_name][0]
+                return person
+        except ValueError as e:
+            cprint('Error', 'red')
```

Comment:
`person = [person for person in amity.all_people if person.person == person_name]`
`if !person: cprint('Error', 'red') `
`else: return person`